src/setup_payload/BUILD.gn

```diff
@@ -14,15 +14,26 @@
 
 import("//build_overrides/chip.gni")
 
-static_library("setup_payload") {
-  output_name = "libSetupPayload"
-
+source_set("additional_data_payload") {
```

Comment:
I took a look at several examples in our sdk like: https://github.com/project-chip/connectedhomeip/blob/master/src/lib/shell/BUILD.gn#L20 or https://github.com/project-chip/connectedhomeip/blob/master/src/lib/dnssd/BUILD.gn#L19 and I thought this is the preferred way to do that. Do you think it would be better to use library instead?

Reply:
@mspang is there a material difference? 

There is https://gn.googlesource.com/gn/+/refs/heads/main/docs/style_guide.md#source-sets-versus-static-libraries mentions some performance benefit for  static libraries, but also says source_sets are almost never wrong .. so unsure.

I  just want to ensure that circular dependencies are avoided.